api/v1alpha1/hostedcluster_types.go

```diff
@@ -267,6 +269,51 @@ type ClusterAutoscaling struct {
 	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
 }
 
+// EtcdManagementType is a enum specifying the strategy for managing the clusters etcd instance
+type EtcdManagementType string
+
+const (
+	Managed   EtcdManagementType = "Managed"
+	Unmanaged EtcdManagementType = "Unmanaged"
+)
+
+type EtcdSpec struct {
+	// ManagementType defines how the etcd cluster is managed. Unmanaged means
+	// the etcd cluster is managed by a system outside the hypershift controllers.
+	// Managed means the hypershift controllers manage the provisioning of the etcd cluster
+	// and the operations around it
+	ManagementType EtcdManagementType `json:"managementType"`
+
+	// Managed provides metadata that defines how the hypershift controllers manage the etcd cluster
+	Managed *ManagedEtcdSpec `json:"managed,omitempty"`
+
+	// Unmanaged provides metadata that enables the Openshift controllers to connect to the external etcd cluster
+	Unmanaged *UnmanagedEtcdSpec `json:"unmanaged,omitempty"`
+}
+
+type ManagedEtcdSpec struct {
+
+	//TODO: Ultimately backup policies, etc can be defined here.
+}
+
+type UnmanagedEtcdSpec struct {
+	// Endpoint is the full url to connect to the etcd cluster endpoint. An example is
+	// https://etcd-client:2379
+	Endpoint string `json:"endpoint"`
```

Comment:
Need to clarify and enforce validation for the contents here, is it really a URL, or is it an address, address:port pair, etc.

Reply:
adding validation

Reply:
Might want to throw in an explicit `// +kubebuilder:validation:Required` while you're at it

Reply:
done